osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -102,44 +152,40 @@ public void ApplyToBeatmap(IBeatmap beatmap)
         }
 
         /// <summary>
-        /// Returns the final position of the hit object
+        /// Get the absolute angle of a vector pointing from the previous hit object to the one denoted by <paramref name="hitObjectIndex"/>.
         /// </summary>
-        /// <returns>Final position of the hit object</returns>
-        private void applyRandomisation(float rateOfChangeMultiplier, RandomObjectInfo previous, RandomObjectInfo current)
+        /// <param name="hitObjects">A list of all hit objects in the beatmap.</param>
+        /// <param name="hitObjectIndex">The hit object that the vector should point to.</param>
+        /// <returns>The absolute angle of the aforementioned vector.</returns>
+        private float getAbsoluteAngle(IReadOnlyList<OsuHitObject> hitObjects, int hitObjectIndex)
         {
-            if (previous == null)
-            {
-                var playfieldSize = OsuPlayfield.BASE_SIZE;
+            if (hitObjectIndex < 0) return 0;
```

Comment:
This method is weird in places like this - it returns dubious outputs for dubious inputs in order to sorta-kinda work with the methods that depend on its return value. Seeing that it is called a grand total of one time, and the result is immediately passed on to `computeRandomisedPosition()`, it may honestly be better to just merge this method in there? I'm thinking about a signature of

```csharp
private void computeRandomisedPosition(
    RandomObjectInfo current,
    RandomObjectInfo previous,
    RandomObjectInfo? beforePrevious = null);
```

which I *believe* should have everything you need to compute the randomised position of `current`?